### ISSUE TITLE
IOS App crashes using banner ads

### ISSUE THREAD

**Issue by fahad-whizpool** [flutter/flutter#132506]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Go to Screen B
2. Click on go back.
I know most of you suggest it's a similar issue #[https://github.com/flutter/flutter/issues/127489],#[https://github.com/flutter/flutter/issues/128018]
 which is fixed on master channel. I tried it on master channel app didn't crash. But master channel spoiled my UI in my app. **e.g** Blue color turns to purple and there were lot of other UI issues as well on unstable channel(master). Crashes mostly occurs on IPAD (15.7 version) mostly in release mode. Also I have noticed some lagging in navigation as well specially on Ipad while using these banner ads from **google_mobile_ads 3.0.0.** 
**Note: App working fine on Android**

### Expected results

App move back to previous screen.

### Actual results

App crashes sometimes.

### Code sample

<details><summary>Code sample</summary>

```dart
Here is link for code sample https://github.com/fahad-whizpool/adaptive_banner_example
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
Here is stack trace. facing on iOS
[com.whizpool.filelink_issue_3ab948099c1290a77effa6e6447c86c8_crash_session_aaca7c6607d147149f814577d345c7d3_DNE_0_v2_stacktrace.txt](https://github.com/googleads/googleads-mobile-flutter/files/12227449/com.whizpool.filelink_issue_3ab948099c1290a77effa6e6447c86c8_crash_session_aaca7c6607d147149f814577d345c7d3_DNE_0_v2_stacktrace.txt](url)

</details>


### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.6, on macOS 13.5 22G74 darwin-x64, locale en-PK)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.81.1)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by stuartmorgan** [comment]
> I tried it on master channel app didn't crash. But master channel spoiled my UI in my app.

If it's already fixed on master, there's nothing for us to track here; fixes are cherry-picked to earlier releases [only in exceptional cases](https://github.com/flutter/flutter/wiki/Flutter-Cherrypick-Process). Given that, closing.

> **e.g** Blue color turns to purple and there were lot of other UI issues as well on unstable channel(master).

Please files issues for any issues you have when using `master`.

**Comment by fahad-whizpool** [comment]
anyone can confirm is this issue fixed in Flutter 3.13 (stable channel)?
